src/pages/MoviePage/MoviePage.tsx

```diff
@@ -1,46 +1,69 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Chip, Typography } from '@mui/material';
+import { useParams } from 'react-router-dom';
 // import classNames from 'classnames';
-import style from './MoviePage.module.scss';
-// import dataMovies from '../../mockedData/data-movies.json';
-import dataMovie from '../../mockedData/data-movie.json';
-import dataImages from '../../mockedData/data-images.json';
-// import Card from '../../components/Card';
+import Card from '../../components/Card';
 import Paragraph from '../../components/Paragraph';
-import { TypeDataMovie } from '../../components/types';
 import ButtonElem from '../../components/ButtonElem';
+import { TypeMoviePage } from '../../components/types';
+import { useAppDispatch, useAppSelector } from '../../store/store';
+import { fetchAllDataMovie } from '../../store/reducers/MovieActions';
+import style from './MoviePage.module.scss';
+import { fetchGenresData } from '../../store/reducers/CardsActions';
 
 function MoviePage() {
-  const { original_title, poster_path, overview, release_date, revenue, runtime, genres } =
-    dataMovie as TypeDataMovie;
   const [isOpenActors, setIsOpenActors] = useState(false);
+  const dispatch = useAppDispatch();
+  const { id } = useParams();
   // const actorsClassNames = classNames(style.actors__wrapper, {
   //   [style.actors__active]: isOpenActors,
   // });
-
   const isOpenActorsCollection = () => {
     setIsOpenActors(() => !isOpenActors);
   };
 
+  const { data, images, recommendations, loading } = useAppSelector((state) => state.movie) || {};
+  const { genresArray } = useAppSelector((state) => state.genres);
+
+  useEffect(() => {
+    dispatch(fetchAllDataMovie(Number(id)));
+    if (genresArray.length === 0) dispatch(fetchGenresData());
+  }, [id, genresArray.length, dispatch]);
+
+  if (!data || loading) return <div>Loading...</div>;
+
+  const { poster_path, title, overview, release_date, revenue, runtime, genres } =
+    data as TypeMoviePage;
+
+  const getTimeFromMins = (mins: number): string => {
```

Comment:
Please create a helpers.ts and move there all general functions. It will be easy to test and support.